### PR TITLE
[emasamplerate] Fix inaccuracies with low throughputs/small counts

### PR DESCRIPTION
**Pull request by tredman**
Adapting the AvgSampleRate implementation, which uses integers, to the Exponential Moving Average, which uses floats, introduced some inaccuracies with very small counts, which end up in the EMA at < 1. This results in their log being < 0, and skews `logSum` to be smaller than it would have been computed in the AvgSampleRate implementation. All this adds up to less accurate sample rate calculations when counts are small or there are a lot of keys with counts < 1. This change puts a floor on the EMA counts at 1 for the purposes of doing the log curve mapping.

I added two tests:
- TestEMASampleUpdateMapsSparseCounts
- TestAvgSampleUpdateMapsSparseCounts

These should be equal, but before my EMASampleRate changes, the EMA test got a sample rate of 5 while the AvgSampleRate test got a sample rate value of 16.

Note that no changes occurred in the expected output of `TestEMASampleUpdateMaps`, which further suggests this was an edge case only affecting small counts.

### CODE REVIEW

File: emasamplerate.go [hunk 1]
```diff
@@ -184,7 +184,10 @@ func (e *EMASampleRate) updateMaps() {
 	// determines what percentage of the total event space belongs to each key
 	var logSum float64
 	for _, count := range e.movingAverage {
-		logSum += math.Log10(float64(count))
+		// We take the max of (1, count) because count * weight is < 1 for
+		// very small counts, which throws off the logSum and can cause
+		// incorrect samples rates to be computed when throughput is low
+		logSum += math.Log10(math.Max(1, count))
```
Comment (tredman) [inline emasamplerate.go]: with enough keys at count = 1 (< 1 in the EMA), logSum could go negative here!

File: emasamplerate.go [hunk 2]
```diff
@@ -205,7 +208,7 @@ func (e *EMASampleRate) updateMaps() {
 	keysRemaining := len(e.movingAverage)
 	var extra float64
 	for _, key := range keys {
-		count := e.movingAverage[key]
+		count := math.Max(1, e.movingAverage[key])
 		// take the max of 1 or my log10 share of the total
 		goalForKey := math.Max(1, math.Log10(count)*goalRatio)
```
Comment (tredman) [inline emasamplerate.go]: this was the only thing saving us from negative goalRatios
